pkg/apis/core/v1beta1/types_shoot.go

```diff
@@ -411,6 +411,9 @@ type ClusterAutoscaler struct {
 	// MaxGracefulTerminationSeconds is the number of seconds CA waits for pod termination when trying to scale down a node (default: 600).
 	// +optional
 	MaxGracefulTerminationSeconds *int32 `json:"maxGracefulTerminationSeconds,omitempty" protobuf:"varint,9,opt,name=maxGracefulTerminationSeconds"`
+	// IgnoreTaints specifies a list of taint keys to ignore in node templates when considering to scale a node group.
+	// +optional
+	IgnoreTaints []string `json:"ignoreTaints,omitempty" protobuf:"bytes,10,opt,name=ignoreTaints"`
```

Comment:
Does it make sense to add some validation for this field? For example preventing that duplicate keys can be specified?

Reply:
Yes, Rafael. I indeed was looking for this suggestion as mentioned in `Special notes for reviewers`. Thank you for this. I will add this validation.

Reply:
@rfranzke I checked the CA's code before adding a validation here. In it, CA creates a set of taintKeys (which will only have unique keys). See [here](https://github.com/gardener/autoscaler/blob/97636aa0a5f09d2cb5c1d6125111d5c760f16140/cluster-autoscaler/core/static_autoscaler.go#L142-L146). So, does it make sense to add such a validation at g/g? I feel it is redundant. WDYT?

Reply:
I don't have a strong opinion, but given that this validation code in CA could change or disappear in future versions and specifying duplicate taint keys seems never to be a valid use case, I would probably still do it to "keep the shoot API clean".

Reply:
Got it. Thanks!